haystack_experimental/components/tools/openapi/_openapi.py

```diff
@@ -0,0 +1,557 @@
+            )
+
+        self.credentials = credentials
+        self.request_sender = request_sender or send_request
+        self.llm_provider: LLMProvider = llm_provider or LLMProvider.OPENAI
+
+    def get_auth_function(self) -> Callable[[Dict[str, Any], Dict[str, Any]], Any]:
+        """
+        Get the authentication function that sets a schema specified authentication to the request.
+
+        The function takes a security scheme and a request as arguments:
+            `security_scheme: Dict[str, Any] - The security scheme from the OpenAPI spec.`
+            `request: Dict[str, Any] - The request to apply the authentication to.`
+        :return: The authentication function.
+        """
+        security_schemes = self.openapi_spec.get_security_schemes()
+        if not self.credentials:
+            return lambda security_scheme, request: None  # No-op function
+        if isinstance(self.credentials, str):
+            return self._create_authentication_from_string(
+                self.credentials, security_schemes
+            )
+        raise ValueError(f"Unsupported credentials type: {type(self.credentials)}")
+
+    def get_tools_definitions(self) -> List[Dict[str, Any]]:
+        """
+        Get the tools definitions used as tools LLM parameter.
+
+        :return: The tools definitions passed to the LLM as tools parameter.
+        """
+        provider_to_converter = {
+            "anthropic": anthropic_converter,
+            "cohere": cohere_converter,
+        }
+        converter = provider_to_converter.get(self.llm_provider.value, openai_converter)
+        return converter(self.openapi_spec)
+
+    def get_payload_extractor(self):
+        """
+        Get the payload extractor for the LLM provider.
+
+        This function knows how to extract the exact function payload from the LLM generated function calling payload.
+        :return: The payload extractor function.
+        """
+        provider_to_arguments_field_name = {
+            "anthropic": "input",
+            "cohere": "parameters",
+        }  # add more providers here
+        # default to OpenAI "arguments"
+        arguments_field_name = provider_to_arguments_field_name.get(
+            self.llm_provider.value, "arguments"
+        )
+        return create_function_payload_extractor(arguments_field_name)
+
+    def _create_authentication_from_string(
+        self, credentials: str, security_schemes: Dict[str, Any]
+    ) -> Callable[[Dict[str, Any], Dict[str, Any]], Any]:
+        for scheme in security_schemes.values():
+            if scheme["type"] == "apiKey":
+                return create_api_key_auth_function(api_key=credentials)
+            if scheme["type"] == "http":
+                return create_http_auth_function(token=credentials)
+            if scheme["type"] == "oauth2":
+                raise NotImplementedError("OAuth2 authentication is not yet supported.")
```

Comment:
Since we only support HTTP and API key, let's just raise for all other authentication types (and print the type in the error message).